packages/component-library/src/components/plmg-tooltip/plmg-tooltip.tsx

```diff
@@ -0,0 +1,270 @@
+
+  @Prop() arrowPosition: arrowPositions = 'middle';
+  @Watch('arrowPosition')
+  validateArrowPosition(newValue: string) {
+    if (newValue && typeof newValue !== 'string')
+      throw new Error('arrow position must be a string');
+    if (!isArrowPosition(newValue))
+      throw new Error('arrow postion must be a valid value');
+  }
+
+  /**
+   * Tooltip Title Text
+   *
+   * Allowed value: any string
+   *
+   * Required
+   */
+
+  @Prop() content: string;
+  @Watch('content')
+  validatetooltipTitle(newValue: string) {
+    if (newValue && !newValue) throw new Error('tooltip requires content');
+    if (newValue && typeof newValue !== 'string')
+      throw new Error('text must be a string');
+  }
+
+  /** Life Cycle Methods & Event Listeners
+   *
+   * Listen for
+   *
+   * mouse over
+   * focus
+   * mouse out
+   * blur
+   *
+   * on the target element
+   *
+   * forceVisibile prop disables the listeners
+   */
+
+  connectedCallback() {
+    this.isVisible = this.forceVisible;
+    if (!this.forceVisible && this.targetElement) {
+      this.targetHTMLElement = document.getElementById(this.targetElement);
+      if (this.targetHTMLElement != null) {
+        this.abortTooltipListener = new AbortController();
+        this.targetHTMLElement.addEventListener(
+          'mouseover',
+          () => (this.isVisible = true),
+          { signal: this.abortTooltipListener.signal }
+        );
+        this.targetHTMLElement.addEventListener(
+          'focus',
+          () => (this.isVisible = true),
+          { signal: this.abortTooltipListener.signal }
+        );
+        this.targetHTMLElement.addEventListener(
+          'mouseleave',
+          () => (this.isVisible = false),
+          { signal: this.abortTooltipListener.signal }
+        );
+        this.targetHTMLElement.addEventListener(
+          'blur',
+          () => (this.isVisible = false),
+          { signal: this.abortTooltipListener.signal }
+        );
+      }
+    }
+  }
+
+  /**
+   * Remove event listener
+   */
+
+  disconnectedCallback() {
+    if (!this.forceVisible) {
+      if (this.abortTooltipListener) this.abortTooltipListener.abort();
+    }
+  }
+
+  render() {
+    const classes = {
+      'plmg-tooltip': true,
+      visible: this.isVisible,
+      // Include arrow classes if side is not set to none
+      ...(this.arrowSide && { [this.arrowSide]: true }),
+      ...(this.arrowSide && { [this.arrowPosition]: true }),
+      [this.backgroundColor]: true,
+    };
+
+    return (
+      <div style={!this.forceVisible && this.setPosition()}>
+        <span class={classes}>{this.content}</span>
+      </div>
+    );
+  }
+
+  private setPosition() {
+    let styles = {
+      position: 'fixed',
+      overflow: 'visible',
+      left: '10%',
+      top: '10%',
+    };
+
+    if (!this.targetHTMLElement) return styles;
+
+    /**
+     * Max width is 150px plus 6px for the arrow = 156px
+     * Max numbers of characters per line is 27 characters
```

Comment:
This is key to be included in the documentation as any word greater than 27 characters will overflow.
<img width="190" alt="Screenshot 2022-05-31 at 12 10 39" src="https://user-images.githubusercontent.com/14997231/171150100-61599cda-2fa1-4b3c-a29a-bba622c2d54e.png">

Reply:
Hadn't thought of that. I've added overflow-wrap: break-word; ![Screenshot 2022-05-31 at 14 36 15](https://user-images.githubusercontent.com/104433221/171175475-f5e6329b-cdb6-4734-b806-20026d4e31d8.png)